frontend/src/pages/NewTag/Form.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { ArrowLeftS } from 'styled-icons/remix-fill/ArrowLeftS';
+
+import { useGetTags } from '../../sharedComponents/hooks/GetTags';
+import { LoadingButton } from '../../sharedComponents/LoadingButton';
+import { alertActions } from '../../state/actions/alert';
+
+const Wrapper = styled.div`
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  background-color: #f9f9f9;
+`;
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-content: space-around;
+  max-width: 30em;
+  min-width: 20em;
+  padding: 1em;
+  margin: 10px 0px;
+  background-color: white;
+  border-radius: 4px;
+  -webkit-box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+  -moz-box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+  box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+`;
+
+const TagsLink = styled(Link)`
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+  width: fit-content;
+  margin-bottom: 10px;
+`;
+
+const StyledInput = styled.input`
+  flex: 0 0 2em;
+  margin-bottom: 5px;
+  font-size: 16px;
+  border-radius: 4px;
+  color: black;
+  border: ${(props) => (props.valid ? '1px solid springgreen' : '1px solid orangered')};
+  border: ${(props) => (props.empty ? '1px solid darkgray' : null)};
+  outline-color: transparent;
+  outline-style: none;
+`;
+
+const Form = () => {
+  // Router
+  const history = useHistory();
+
+  // Redux
+  const dispatch = useDispatch();
+  const userSelector = useSelector((state) => state.user);
+  const { token } = userSelector.user;
+
+  // State
+  const allTags = useGetTags() || [];
+  const [tag, setTag] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [valid, setValid] = useState(false);
+
+  // Update input teext
+  const onChange = (e) => {
+    setTag(e.target.value);
+  };
+
+  // Only allow tags with letters
+  const validateTag = (_tag) => {
+    const re = /^[a-åA-Å\s]*$/;
+    return re.test(_tag);
+  };
+
+  // Check if the tag already exists
+  const checkIfTagExists = (_tag) => {
+    for (const exsistingTag of allTags) {
+      if (exsistingTag.toLowerCase() === _tag.toLowerCase()) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  // Check if the tag exists and follows the correct pattern and
+  // sets the border color according to tat
+  const checkTag = () => {
+    if (!validateTag(tag) || checkIfTagExists(tag)) {
+      setValid(false);
+    } else {
+      setValid(true);
+    }
+  };
```

Comment:
Seems like what this is doing is just
```suggestion
  const checkTag = () => {
    setValid(validateTag(tag) && !checkIfTagExists(tag));
  };
```

Reply:
Makes readability worse, but idm if we wanna save the 4 lines.